test/boolean_complex_test.cpp

```diff
@@ -1030,4 +1031,159 @@ TEST(BooleanComplex, HullMask) {
+    e32 = (eind[0] < eind[2]) ? eind[2] : eind[0];
+    edges.insert(std::make_pair(e11, e12));
+    edges.insert(std::make_pair(e21, e22));
+    edges.insert(std::make_pair(e31, e32));
+  }
+  manifold::Manifold c;
+  // Vertex Spheres
+  Manifold sph = Manifold::Sphere(1, 8);
+  for (size_t i = 0; i < seeds.NumVert(); i++) {
+    vec3 vpos(seeds.vertProperties[3 * i + 0], seeds.vertProperties[3 * i + 1],
+              seeds.vertProperties[3 * i + 2]);
+    Manifold vsph = sph.Translate(vpos);
+    manifold::Manifold left = c;
+    manifold::Manifold right(vsph);
+    if (!right.NumTri()) continue;
+    c = left.Boolean(right, manifold::OpType::Add);
+    EXPECT_TRUE(c.NumTri() > 0);
+  }
+  // Edge Cylinders
+  std::set<std::pair<int, int>>::iterator e_it;
+  for (e_it = edges.begin(); e_it != edges.end(); ++e_it) {
+    vec3 ev1 = vec3(seeds.vertProperties[3 * e_it->first + 0],
+                    seeds.vertProperties[3 * e_it->first + 1],
+                    seeds.vertProperties[3 * e_it->first + 2]);
+    vec3 ev2 = vec3(seeds.vertProperties[3 * e_it->second + 0],
+                    seeds.vertProperties[3 * e_it->second + 1],
+                    seeds.vertProperties[3 * e_it->second + 2]);
+    vec3 edge = ev2 - ev1;
+    double len = la::length(edge);
+    if (len < std::numeric_limits<float>::min()) continue;
+    // TODO - workaround, shouldn't be necessary
+    if (len < 0.03) continue;
+    manifold::Manifold origin_cyl = manifold::Manifold::Cylinder(len, 1, 1, 8);
+    vec3 evec(-1 * edge.x, -1 * edge.y, edge.z);
+    manifold::Manifold rotated_cyl =
+        origin_cyl.Transform(manifold::RotateUp(evec));
+    manifold::Manifold right = rotated_cyl.Translate(ev1);
+    if (!right.NumTri()) continue;
+    manifold::Manifold left = c;
+    c = left.Boolean(right, manifold::OpType::Add);
+    EXPECT_TRUE(c.NumTri() > 0);
+  }
+  // Triangle Volumes
+  for (size_t i = 0; i < seeds.NumTri(); i++) {
+    int eind[3];
+    for (int j = 0; j < 3; j++) eind[j] = seeds.triVerts[i * 3 + j];
+    vec3 ev1 = vec3(seeds.vertProperties[3 * eind[0] + 0],
+                    seeds.vertProperties[3 * eind[0] + 1],
+                    seeds.vertProperties[3 * eind[0] + 2]);
+    vec3 ev2 = vec3(seeds.vertProperties[3 * eind[1] + 0],
+                    seeds.vertProperties[3 * eind[1] + 1],
+                    seeds.vertProperties[3 * eind[1] + 2]);
+    vec3 ev3 = vec3(seeds.vertProperties[3 * eind[2] + 0],
+                    seeds.vertProperties[3 * eind[2] + 1],
+                    seeds.vertProperties[3 * eind[2] + 2]);
+    vec3 a = ev1 - ev3;
+    vec3 b = ev2 - ev3;
+    vec3 n = la::normalize(la::cross(a, b));
+    // Extrude the points above and below the plane of the triangle
+    vec3 pnts[6];
+    pnts[0] = ev1 + n;
+    pnts[1] = ev2 + n;
+    pnts[2] = ev3 + n;
+    pnts[3] = ev1 - n;
+    pnts[4] = ev2 - n;
+    pnts[5] = ev3 - n;
+    // Construct the points and faces of the new manifold
+    double pts[3 * 6];
```

Comment:
Mind converting this to an initializer list so it's not quite so verbose? e.g.
```
double pts[3 * 6]={pnts[4].x, pnts[4].y, pnts[4].z,\\
                              ...
                             };
```

Reply:
Updated.  I'm not quite following the Warp + Extrude logic yet, so my initial attempt there didn't work - I'll have to look at it next week unless this makes things readable enough.

Reply:
Thanks, that's loads better - you're welcome to leave it. I guess go ahead and put your status checks back the way they were so we at least have something to test. Maybe leave these ones commented out with some info on how they caused the triangulator problem.